packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,725 @@
+               int32_t maxWidth) {
+  // Sort boxes descending by maximum side length
+  std::sort(boxes.begin(), boxes.end(),
+            [](const DetectorBBox &lhs, const DetectorBBox &rhs) {
+              return maxSideLength(lhs.bbox) > maxSideLength(rhs.bbox);
+            });
+
+  std::vector<DetectorBBox> mergedVec;
+  float lineAngle;
+  while (!boxes.empty()) {
+    auto currentBox = boxes[0];
+    float normalizedAngle = normalizeAngle(currentBox.angle);
+    boxes.erase(boxes.begin());
+    std::unordered_set<std::size_t> ignoredIdxs;
+
+    while (true) {
+      // Find all aligned boxes and merge them until max_size is reached or no
+      // more boxes can be merged
+      auto [slope, intercept, isVertical] =
+          fitLineToShortestSides(currentBox.bbox);
+
+      lineAngle = std::atan(slope) * 180 / M_PI;
+      if (isVertical) {
+        lineAngle = -90;
+      }
+
+      auto closestBoxInfo =
+          findClosestBox(boxes, ignoredIdxs, currentBox.bbox, isVertical, slope,
+                         intercept, centerThreshold);
+      if (!closestBoxInfo)
+        break;
+
+      std::size_t candidateIdx = closestBoxInfo.value().first;
+      DetectorBBox candidateBox = boxes[candidateIdx];
+      const float candidateHeight = closestBoxInfo.value().second;
+
+      if ((candidateBox.angle == 90 && !isVertical) ||
+          (candidateBox.angle == 0 && isVertical)) {
+        candidateBox.bbox = rotateBox(candidateBox.bbox, normalizedAngle);
+      }
+
+      const float minDistance = calculateMinimalDistanceBetweenBox(
+          candidateBox.bbox, currentBox.bbox);
+      const float mergedHeight = minSideLength(currentBox.bbox);
+      if (minDistance < distanceThreshold * candidateHeight &&
+          std::fabs(mergedHeight - candidateHeight) <
+              candidateHeight * heightThreshold) {
+        currentBox.bbox = mergeRotatedBoxes(currentBox.bbox, candidateBox.bbox);
+        boxes.erase(boxes.begin() + candidateIdx);
+        ignoredIdxs.clear();
+        if (maxSideLength(currentBox.bbox) > maxWidth) {
+          break;
+        }
+      } else {
+        ignoredIdxs.insert(candidateIdx);
+      }
+    }
+
+    mergedVec.emplace_back(currentBox.bbox, lineAngle);
+  }
+
+  // Remove small boxes and sort by vertical
+  mergedVec =
+      removeSmallBoxesFromArray(mergedVec, minSideThreshold, maxSideThreshold);
+
+  std::sort(mergedVec.begin(), mergedVec.end(),
+            [](const auto &obj1, const auto &obj2) {
+              const auto &coords1 = obj1.bbox;
+              const auto &coords2 = obj2.bbox;
+              const float minY1 = minimumYFromBox(coords1);
+              const float minY2 = minimumYFromBox(coords2);
+              return minY1 < minY2;
+            });
+
+  std::vector<DetectorBBox> orderedSortedBoxes;
+  orderedSortedBoxes.reserve(mergedVec.size());
+  for (DetectorBBox bbox : mergedVec) {
+    bbox.bbox = orderPointsClockwise(bbox.bbox);
+    orderedSortedBoxes.push_back(bbox);
```

Comment:
Default moving constructor for `DetectorBBox` works here for us
```suggestion
    orderedSortedBoxes.push_back(std::move(bbox));
```